src/details/ArborX_DetailsDistributor.hpp

```diff
@@ -107,45 +107,29 @@ class Distributor
     sortAndDetermineBufferLayout(destination_ranks, _permute, _destinations,
                                  _dest_counts, _dest_offsets);
 
+    std::vector<int> mask(comm_size);
+    for (int i = 0; i < _destinations.size(); ++i)
     {
-      std::vector<int> mask(comm_size);
-      for (auto i : _destinations)
-        mask[i] = 1;
-      MPI_Alltoall(MPI_IN_PLACE, 0, MPI_DATATYPE_NULL, mask.data(), 1, MPI_INT,
-                   _comm);
-      for (int i = 0; i < comm_size; ++i)
-        if (mask[i] == 1)
-          _sources.push_back(i);
+      // The previous behaviour was that even if _dest_counts[i] == 0, we still
+      // added that to the source lists. Here, we add 1 to retain that, even
+      // though it is weird as of why we would have such cases.
+      mask[_destinations[i]] = _dest_counts[i] + 1;
```

Comment:
What happens if we don't do it? At first sight, this really looks unnecessary.

Reply:
If we remain in the scope of this function, and don't do that, then the senders are going to be different from receivers, and things are going to deadlock somewhere. I'm not sure about ramifications of that, and whether this happens.

Reply:
I see. Fine with me then.